src/Parser/Tokenizer/Tokenizer.php

```diff
@@ -44,7 +44,10 @@ public static function fromSource(Source $source): Tokenizer
      */
     public function getIterator(): \Iterator
     {
-        yield from self::block($this->source->getIterator());
+        $fragments = $this->source->getIterator();
+        while ($fragments->valid()) {
+            yield from self::block($fragments);
+        }
```

Comment:
`while ($fragments->valid())` is imho tautological to `yield from`. At least, when I revert these lines, tests stay green.

What was your reasoning for putting the source iterator into a variable?